### PR TITLE
Show notification for backup running in the background

### PR DESCRIPTION
**Pull request by grote**
The system triggers backup jobs periodically or when a package is announcing that its data has changed. So far we were not showing notifications for those.

This MR shows a notification with an indeterminate progress bar as we don't have any information about how many packages will get backed up.

### CODE REVIEW

File: app/src/main/java/com/stevesoltys/seedvault/metadata/MetadataManager.kt [hunk 1]
```diff
@@ -198,6 +198,8 @@ class MetadataManager(
 
     @Synchronized
     fun getPackagesNumBackedUp(): Int {
+        // FIXME we are under-reporting packages here,
+        //  because we have no way to also include upgraded system apps
```
Comment (chirayudesai) [inline app/src/main/java/com/stevesoltys/seedvault/metadata/MetadataManager.kt:202]: spacing

File: app/src/main/java/com/stevesoltys/seedvault/settings/SettingsViewModel.kt [hunk 2]
```diff
@@ -65,7 +69,11 @@ internal class SettingsViewModel(
     }
 
     internal fun backupNow() {
-        Thread { requestBackup(app) }.start()
+        if (notificationManager.hasActiveBackupNotifications()) {
+            Toast.makeText(app, R.string.notification_backup_already_running, LENGTH_LONG).show()
```
Comment (chirayudesai) [inline app/src/main/java/com/stevesoltys/seedvault/settings/SettingsViewModel.kt:73]: I'm not the biggest fan of toast messages tbh.

A snackbar might be better.

File: app/src/main/java/com/stevesoltys/seedvault/ui/notification/BackupNotificationManager.kt [hunk 3]
```diff
@@ -16,6 +16,7 @@ import androidx.core.app.NotificationCompat.Builder
 import androidx.core.app.NotificationCompat.PRIORITY_DEFAULT
 import androidx.core.app.NotificationCompat.PRIORITY_HIGH
 import androidx.core.app.NotificationCompat.PRIORITY_LOW
+import com.stevesoltys.seedvault.BuildConfig.APPLICATION_ID
```
Comment (chirayudesai) [inline app/src/main/java/com/stevesoltys/seedvault/ui/notification/BackupNotificationManager.kt:19]: This won't be available when building in AOSP, without gradle.

Looks like it's already in the other notification commit too - https://github.com/stevesoltys/seedvault/pull/111 - that'll break build.

**Review by chirayudesai — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/stevesoltys/seedvault/metadata/MetadataManager.kt (hunk 1 above)
Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/metadata/MetadataManager.kt:202]: The spacing looks indeed odd, but it is intentional. This way the IDE highlights the multi line comment as FIXME

File: app/src/main/java/com/stevesoltys/seedvault/settings/SettingsViewModel.kt (hunk 2 above)
Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/settings/SettingsViewModel.kt:73]: We are in a view model here and can't show a snackbar. This is just to give some sort of feedback why we won't run a backup now and under normal circumstances, this won't be triggered (only when a user wants to manually trigger a backup, but didn't notice that one is already running). Not sure the extra effort for a snackbar would be justified.

**Comment by chirayudesai** [comment]
Some alternatives in case `BuildConfig` is needed for other things in the future:
* https://cs.android.com/search?q=BuildConfig.java&sq=
* https://github.com/LineageOS/android_packages_apps_Eleven/commit/aff7e1d73f82ab2d8dfb1a08ef469db32422fb8c

Basically just commit a BuildConfig.java for aosp build in a separate dir

**Review by chirayudesai — APPROVED** [review]

**Review by stevesoltys — APPROVED** [review]

**Pull request merged**